src/components/FilterPanel/FilterPanelCheckboxWithOverflowMenu/FilterPanelCheckboxWithOverflowMenu.js

```diff
@@ -0,0 +1,119 @@
+/**
+ * @file Filter panel checkbox with overflow menu component.
+ * @copyright IBM Security 2020
+ */
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import classnames from 'classnames';
+
+import { getComponentNamespace } from '../../../globals/namespace';
+
+import FilterPanelCheckbox from '../FilterPanelCheckbox';
+import OverflowMenu from '../../OverflowMenu';
+
+export const namespace = getComponentNamespace(
+  'filter-panel-checkbox-with-overflow-menu'
+);
+
+const FilterPanelCheckboxWithOverflowMenu = ({
+  className,
+  checkboxClassName,
+  checkboxWrapperClassName,
+  overflowMenuAriaLabel,
+  overflowMenuClassName,
+  overflowMenuOptionsClassName,
+  children,
+  ...other
+}) => {
+  const [showOverflowButton, setShowOverflowButton] = React.useState(false);
+  const [onBlurTimeout, setOnBlurTimeout] = React.useState(undefined);
+
+  // Clear any timeouts before unmounting.
+  React.useEffect(() => () => clearTimeout(onBlurTimeout), [onBlurTimeout]);
+  const menuOptionsClassName = `${namespace}__overflow-options`;
+
+  // Set a short delay to detect if the blur event resulted in a complete loss of keyboard focus or
+  // the focus just shifted to another element within this same component.
+  const handleBlur = event => {
```

Comment:
Hey @diego-codes! I've implemented a very similar component on Falcon and run into this same issue. I used something like this snippet:

```
  let timeout = null;

  const handleFocus = () => {
    clearTimeout(timeout);
    if (!setShowOverflowButton) {
      setShowOverflowButton(true);
    }
  };

  const handleBlur = () => {
    timeout = setTimeout(() => {
      if (showOverflowButton) {
        setShowOverflowButton(false);
      }
    }, 10);
  };
```

It basically relies on the `onFocus` event from the next descendant receiving focus to bubble up straight after `onBlur` and clear the `handleBlur` timeout before it can execute. It's a bit less code  and no use of `document` but also relies on the descendants allowing it's `onFocus` event to propagate up so perhaps your method is more robust!

Reply:
Hi @ShayMurnin! Thanks for the code snippet. I like it! What elements did you apply the `handleFocus` and `handleBlur` handlers to? I'm wondering because the issue I ran into was that the overflow menu options container does not share the same DOM hierarchy as the overflow menu button.

Reply:
Okay, I got your snippet to work!! And it makes a good candidate for a hook that can also be used for the `FilterPanelSearch`. Thank you so much, @ShayMurnin! Super clean approach ✨